src/azure-cli/azure/cli/command_modules/servicebus/custom.py

```diff
@@ -436,30 +419,35 @@ def cli_migration_start(client, resource_group_name, namespace_name, target_name
 timedeltapattern = re.compile("^\\d+:\\d+:\\d+$")
 
 
-def return_valid_duration(objinstance, duration_property, update_value):
+def return_valid_duration(instance_value, update_value):
     from datetime import timedelta
     from isodate import parse_duration
     if update_value is not None:
-        if iso8601pattern.match(update_value):
-            for attr, value in vars(objinstance).items():
-                if attr == duration_property:
-                    if parse_duration(update_value) < timedelta(days=10675199, minutes=10085, seconds=477581):
-                        return update_value
+        value_toreturn = update_value
+    else:
+        value_toreturn = str(instance_value)
 
-                    if parse_duration(update_value) > timedelta(days=10675199, minutes=10085, seconds=477581):
-                        return None
+    if iso8601pattern.match(value_toreturn):
+        if parse_duration(value_toreturn) < timedelta(days=10675199, minutes=10085, seconds=477581):
+            return value_toreturn
 
-        if timedeltapattern.match(update_value):
-            day, minute, seconds = update_value.split(":")
-            for attr, value in vars(objinstance).items():
-                if attr == duration_property:
-                    if timedelta(days=int(day), minutes=int(minute), seconds=int(seconds)) < timedelta(days=10675199, minutes=10085, seconds=477581):
-                        return timedelta(days=int(day), minutes=int(minute), seconds=int(seconds))
+        if parse_duration(value_toreturn) > timedelta(days=10675198, minutes=10085, seconds=477581):
```

Comment:
Why is this value not the same as the one on line 431?  I'm assuming that (days=10675199, minutes=10085, seconds=477581) is the max duration.

Reply:
fixed